probe/probe.go

```diff
@@ -225,25 +228,44 @@ ForLoop:
 			t.Controls = report.Controls{}
 		})
 	}
-	if err := p.publisher.Publish(rpt); err != nil {
-		log.Infof("Publish: %v", err)
-	}
+	return rpt
 }
 
 func (p *Probe) publishLoop() {
 	defer p.done.Done()
 	pubTick := time.Tick(p.publishInterval)
+	publishCount := 0
+	var lastFullReport report.Report
 
 	for {
+		var err error
 		select {
 		case <-pubTick:
-			p.drainAndPublish(report.MakeReport(), p.spiedReports)
+			rpt := p.drainAndSanitise(report.MakeReport(), p.spiedReports)
+			fullReport := (publishCount % p.ticksPerFullReport) == 0
+			if !fullReport {
+				rpt.UnsafeUnMerge(lastFullReport)
+			}
+			err = p.publisher.Publish(rpt)
+			if err == nil {
+				if fullReport {
+					lastFullReport = rpt
+				}
```

Comment:
Interesting point.  My thinking was that this way it would cope better with one or two delta reports getting lost, although the way the error check is coded makes it fall back to full report if that does happen. 

In practice Scope data doesn't change slowly over time (except metrics, which we deal with differently).

I'll have a think about it.

Reply:
I'm going to leave this unchanged as I find it easier to reason about the current design.